src/components/BookingModal/BookingModalForm/container.js

```diff
@@ -28,48 +31,69 @@ const Container = Wrapped =>
       bookingDuration: PT.number,
       createEvent: PT.func.isRequired,
       updateEvent: PT.func.isRequired,
-      isSameDay: PT.bool.isRequired,
+      toggleRejectionMessageView: PT.bool.isRequired,
+      onFormUpdate: PT.func.isRequired,
+      hasAvailableDays: PT.bool.isRequired,
+      totalHolidays: PT.number.isRequired,
+      availableDays: PT.number.isRequired,
+      workingFromHomeBooking: PT.bool.isRequired,
     };
-
     static defaultProps = {
       isEventBeingUpdated: false,
       bookingDuration: 1,
     };
-
     constructor(props) {
       super(props);
-      const { isSameDay } = this.props;
+      const { workingFromHomeBooking } = this.props;
+
       this.state = {
         formData: {
           end: moment(),
-          eventTypeId: isSameDay ? 2 : 1 ,
+          eventTypeId: workingFromHomeBooking ? eventTypes.WFH : eventTypes.ANNUAL_LEAVE,
           isHalfday: false,
           start: moment(),
           employeeRejectionMessage: '',
           updateMessage: '',
         },
+        submitButtonDisabled: false,
         formIsValid: true,
         capturedRejectionReponseText: '',
+        workingFromHomeBooking: this.WFHBooking(),
       };
     }
-
     componentDidMount = () => {
       const {
         booking: { start, end, eventType, halfDay },
-        isSameDay,
+        onFormUpdate,
       } = this.props;
-      this.setState({
-        formData: {
-          start: start,
-          end: end,
-          eventTypeId: isSameDay ? 2 : eventType.eventTypeId,
-          isHalfday: halfDay || false,
-          employeeRejectionMessage: '',
-          updateMessage: '',
+      this.setState(
+        {
+          formData: {
+            start: start,
+            end: end,
+            eventTypeId: this.state.workingFromHomeBooking
+              ? eventTypes.WFH
+              : eventType.eventTypeId,
+            isHalfday: halfDay || false,
+            employeeRejectionMessage: '',
+            updateMessage: '',
+          },
         },
-      });
+        () => {
+          onFormUpdate(this.state.formData);
+        }
+      );
     };
 
+    WFHBooking() {
```

Comment:
Since this is returning a boolean it should be called something like `isWFHBooking()`